cmd/chain-maind/app/app.go

```diff
@@ -143,20 +144,20 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 			}
 		}()
 		var genesis map[string]interface{}
-		if err := json.NewDecoder(file).Decode(&genesis); err != nil {
-			return err
+		if decodeErr := json.NewDecoder(file).Decode(&genesis); decodeErr != nil {
+			return decodeErr
 		}
 
-		if err := mergo.Merge(&genesis, genesisPatch, mergo.WithOverride); err != nil {
-			return err
-		}
-		if err := file.Truncate(0); err != nil {
-			return err
+		if mergeErr := mergo.Merge(&genesis, genesisPatch, mergo.WithOverride); mergeErr != nil {
+			return mergeErr
 		}
-		if _, err := file.Seek(0, 0); err != nil {
```

Comment:
file open can be removed too.

Reply:
From the code I traced, the system will still load the genesis settings from the file (when the file exists, otherwise generate a default one) and then merge with the settings for the CRO network. If the user modified some parameters in the genesis file. It will be kept. Or is it not a nexpected behavior?

Reply:
the explanation is in the following comment